### PR TITLE
util/types: mysqlTime implements TimeInternal interface

### PR DESCRIPTION
**Pull request by tiancaiamao**
a further step towards mysql time compatibility.
mysqlTime implements TimeInternal interface, and it's close to mysql time representation.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM
@coocood PTAL

File: evaluator/builtin_time_test.go [hunk 1]
```diff
@@ -591,9 +591,9 @@ func (s *testEvaluatorSuite) TestStrToDate(c *C) {
 		Success bool
 		Expect  time.Time
 	}{
-		{"20161122165022", "%Y%m%d%H%i%s", true, time.Date(2016, 11, 22, 16, 50, 22, 0, time.UTC)},
-		{"2016 11 22 16 50 22", "%Y%m%d%H%i%s", true, time.Date(2016, 11, 22, 16, 50, 22, 0, time.UTC)},
-		{"16-50-22 2016 11 22", "%H-%i-%s%Y%m%d", true, time.Date(2016, 11, 22, 16, 50, 22, 0, time.UTC)},
+		{"20161122165022", "%Y%m%d%H%i%s", true, time.Date(2016, 11, 22, 16, 50, 22, 0, time.Local)},
```
Comment (coocood) [inline evaluator/builtin_time_test.go]: Why change UTC to Local? Does it fail when use `UTC`?

Comment (tiancaiamao) [inline evaluator/builtin_time_test.go]: Yes, it fails...
Support for `str_to_date` is still incomplete, after replace the time representation, I'll go to details, do necessary clean up and add more tests, as I've said before.

**Comment by coocood** [comment]
LGTM

**Pull request merged**